plugins/BEdita/Core/src/Model/Entity/ObjectType.php

```diff
@@ -247,4 +248,40 @@ protected function _setParentName($parentName)
 
         return $parentName;
     }
+
+    /**
+     * Getter for virtual property `schema`.
+     *
+     * @return mixed
+     */
+    protected function _getSchema()
+    {
+        if ($this->is_abstract || empty($this->id)) {
+            return false;
+        }
+
+        /** @var \BEdita\Core\Model\Entity\Property[] $allProperties */
+        $allProperties = TableRegistry::get('Properties')
+            ->find('objectType', [$this->id])
+            ->toArray();
+        $entity = TableRegistry::get($this->name)->newEntity();
+        $hiddenProperties = $entity->hiddenProperties();
+
+        $properties = $required = [];
+        foreach ($allProperties as $property) {
+            $accessMode = null;
+            if (!$entity->isAccessible($property->name)) {
+                $accessMode = 'readOnly';
+            } elseif (in_array($property->name, $hiddenProperties)) {
+                $accessMode = 'writeOnly';
+            }
+            $properties[$property->name] = $property->getSchema($accessMode);
+
+            if ($property->required && $accessMode !== 'readOnly') {
```

Comment:
does this mean that hidden properties could be `required` ? :thinking:

Reply:
If I understood `writeOnly` properties can be traced as `required` but `writeOnly` props never appears in API responses.
Maybe should be 
```
if ($property->required && $accessMode === null)
```
or maybe I miss something :smile_cat:

Reply:
In theory (...) I think that a `writeOnly` property could be required. For instance, even if it is not our case, one shall not create a User without a password, or an Application without an API key, …. Those would be examples of required fields that are writeOnly.

For readOnly fields the concept of required is IMHO less meaningful. However I agree with @batopa that requiring `$accessMode === null` might avoid this confusion.

Reply:
ok @batopa proposal seems fine